src/frontend/fcitx4frontend/fcitx4frontend.cpp

```diff
@@ -253,6 +256,24 @@ class Fcitx4InputContext : public InputContext,
         return keyEvent(event) ? 1 : 0;
     }
 
+    bool isVirtualKeyboardVisibleDBus() const {
```

Comment:
there should be no addition to fcitx4 frontend.

Reply:
Could you explain more?

Reply:
There is no point to change fcitx 4 frontend, fcitx4 is dead code. You can't add new feature to fcitx4.

Reply:
> There is no point to change fcitx 4 frontend, fcitx4 is dead code. You can't add new feature to fcitx4.

So I should remove all the new code in fcitx4 frontend?